ProcessLib/RichardsFlow/RichardsFlowFEM.h

```diff
@@ -159,6 +172,57 @@ class LocalAssemblerData : public RichardsFlowLocalAssemblerInterface
         }  // end of mass lumping
     }
 
+    void computeSecondaryVariableConcrete(
+        double const t, std::vector<double> const& local_x) override
+    {
+        SpatialPosition pos;
+        pos.setElementID(_element.getID());
+
+        auto const K = _process_data.intrinsic_permeability(t, pos)[0];
+        MathLib::PiecewiseLinearInterpolation const& interpolated_Pc =
+            _process_data.interpolated_Pc;
+        MathLib::PiecewiseLinearInterpolation const& interpolated_Kr =
+            _process_data.interpolated_Kr;
+        auto const mu = _process_data.viscosity(t, pos)[0];
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        for (unsigned ip = 0; ip < n_integration_points; ++ip)
+        {
+            auto const& sm = _shape_matrices[ip];
+
+            double P_int_pt = 0.0;
+            NumLib::shapeFunctionInterpolate(local_x, sm.N, P_int_pt);
+            double const Pc = -P_int_pt;
+            double const Sw = interpolated_Pc.getValue(Pc);
+            double const k_rel = interpolated_Kr.getValue(Sw);
+            double const K_mat_coeff = K * k_rel / mu;
+
+            auto const num_nodes = ShapeFunction::NPOINTS;
+            auto p_nodal_values =
```

Comment:
Remove line 202 and use ShapeFunction::NPOINTS directly as
```
const auto  p_nodal_values =  Eigen::Map<const NodalVectorType>(&local_x[0], ShapeFunction::NPOINTS);
```
Besides, move it to a place just ahead of  the  loop of `` for (unsigned ip = 0; ...)``